src/containers/BooksForm.js

```diff
@@ -10,25 +12,55 @@ const categories = [
   'Sci-Fi',
 ];
 
-const BooksForm = () => (
-  <form>
-    <h4> Title:</h4>
-    <input
-      type="text"
-      name="title"
-      value="{title}"
-    />
-    <h4>Category:</h4>
-    <select
-      name="category"
-      value=""
-    >
-      {categories.map((category) => (
-        <option key={category}>{category}</option>
-      ))}
-    </select>
-    <button type="submit">Submit</button>
-  </form>
-);
+const BooksForm = () => {
+  const [book, setBook] = useState({ title: '', category: '' });
+
+  let title = '';
+  let category = '';
+  const dispatch = useDispatch();
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+
+    if (name === 'title') {
+      title = value;
+      setBook({ ...book, title });
+    }
+    if (name === 'category') {
+      category = value;
+      setBook({ ...book, category });
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook({
+      ...book,
+      id: Math.random().toString(36).substring(7),
+    }));
```

Comment:
- [ ] [OPTIONAL]  you should `reset the STATE `once the book is created.  [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleSubmit.png)

Reply:
Thanks, I tried but didn't pull this off. Any tips about it?